submissions/yaripey/memory-pair-game/js/game.js

```diff
@@ -0,0 +1,162 @@
+import { Element, getElementName } from './elements.js'
+import { showArrayOfElements, shuffleArray, selectProperTarget } from './utility.js'
+import { congratulations } from './scenes.js'
+
+// The main game class
+const Game = function () {
+
+  // Stores element that is opened right now
+  this.currentOpened = null
+
+  // Amount of pairs of cards;
+  this.amountOfPairs = 16 / 2;
+
+  // Scene counter, initial value is -1 because the function increments it at first.
+  this.currentScene = -1;
+
+  this.mainContainer = document.querySelector('.container')
+  this.stop = false;
+  this.musicMuted = false;
+}
+
+// Function for the elements array initialisation
+Game.prototype.initiateElements = function () {
+  const preShuffleElements = [];
+  for (let i = 0; i < this.amountOfPairs; i++) {
+    const currentElement = getElementName()
+    const newElem = new Element(currentElement, i)
+    const anotherNewElem = new Element(currentElement, i + this.amountOfPairs)
+    preShuffleElements.push(newElem)
+    preShuffleElements.push(anotherNewElem)
+  }
+  this.elements = shuffleArray(preShuffleElements)
+
+}
+
+// Game initialisation function
+// Also stores the main game logic
+Game.prototype.startGame = function (scenesArray) {
+  this.initiateElements()
+
+  const mainGame = new Scene()
+
+  mainGame.mainBlock = document.createElement('div')
+  mainGame.mainBlock.setAttribute('class', 'game-block')
+  this.elements.forEach(elem => {
+    mainGame.mainBlock.appendChild(elem.makeElement())
+  })
+  mainGame.container.appendChild(mainGame.mainBlock)
+
+  this.mainContainer.addEventListener('click', ({ target }) => {
+    if (this.stop) { return }
+    if (!(target.classList.contains('game') || target.classList.contains('game-block'))) {
+      target = selectProperTarget(target)
+      if (target) {
+        const clickedElement = this.elements.find(elem => {
+          return elem.id === parseInt(target.id)
+        })
```

Comment:
`elem => elem.id === parseInt(....)` or `({id}) => id === parseInt(...)`?
Besides, luckily `parseInt` is a cheap operation. Normally you'd want to pre-save value for potentially tons of comparisons against it.